apps/src/lib/client/utils.rs

```diff
@@ -88,36 +100,38 @@ pub async fn join_network(
     }
     let base_dir_full = fs::canonicalize(base_dir).await.unwrap();
 
-    let release_filename = format!("{}.tar.gz", chain_id);
-    let release_url =
-        format!("{}/{}/{}", RELEASE_PREFIX, chain_id, release_filename);
-    let cwd = env::current_dir().unwrap();
-
-    // Read or download the release archive
-    println!("Downloading config release from {} ...", release_url);
-    let release = match download_file(release_url).await {
-        Ok(contents) => contents,
-        Err(error) => {
-            eprintln!("Error downloading release: {}", error);
-            cli::safe_exit(1);
-        }
-    };
-
-    // Decode and unpack the archive
-    let mut decoder = GzDecoder::new(&release[..]);
-    let mut tar = String::new();
-    decoder.read_to_string(&mut tar).unwrap();
-    let mut archive = tar::Archive::new(tar.as_bytes());
-
     // If the base-dir is non-default, unpack the archive into a temp dir inside
     // first.
+    let cwd = env::current_dir().unwrap();
     let (unpack_dir, non_default_dir) =
         if base_dir_full != cwd.join(config::DEFAULT_BASE_DIR) {
             (base_dir.clone(), true)
         } else {
             (PathBuf::from_str(".").unwrap(), false)
         };
-    archive.unpack(&unpack_dir).unwrap();
+
+    let release_url = chain_tgz_url(DEFAULT_RELEASES_SERVER, &chain_id);
+    println!("Downloading config release from {} ...", release_url);
+    download_and_unpack(&release_url, &unpack_dir)
+        .await
+        .unwrap();
+
+    // MASP params may not be present for every release
+    let masp_url = masp_params_tgz_url(DEFAULT_RELEASES_SERVER, &chain_id);
+    if not_found(&masp_url).await.unwrap() {
```

Comment:
We could just call `download()` straight away and do control flow based on whether a 404 error is returned, but thought an explicit `not_found()` would be cleaner in the code though it does mean an extra HTTP HEAD request (although HEAD requests should be lightweight).